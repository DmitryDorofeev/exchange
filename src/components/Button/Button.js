import React from 'react';
import styles from './Button.module.css';
import classnames from 'classnames';

export const Button = ({children, onClick, disabled}) => {
  return (
    <button
      className={classnames(styles.button, {[styles.disabled]: disabled})}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </button>
  )
};