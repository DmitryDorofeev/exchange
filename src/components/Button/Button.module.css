.button {
    border: 1px solid #ffffff;
    border-radius: 5px;
    background: transparent;
    cursor: pointer;
    color: #ffffff;
    font-size: 20px;
}

.disabled {
    opacity: .5;
    cursor: default;
}