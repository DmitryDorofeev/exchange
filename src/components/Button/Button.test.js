import React from 'react';
import {shallow} from 'enzyme';
import {Button} from './Button';

describe('Button component', () => {
  const button = shallow(<Button/>);

  it('matches snapshot', () => {
    expect(button).toMatchSnapshot();
  });

  it('runs onClick function', () => {
    const mock = jest.fn();

    button.setProps({
      onClick: mock,
    });

    button.simulate('click');

    expect(mock).toBeCalled();
  });
});

describe('Button component disabled', () => {
  const button = shallow(<Button disabled/>);

  it('matches snapshot', () => {
    expect(button).toMatchSnapshot();
  });

});