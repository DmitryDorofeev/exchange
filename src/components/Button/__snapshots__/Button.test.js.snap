// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component disabled matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "button disabled",
      "disabled": true,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "button disabled",
        "disabled": true,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Button component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "button",
      "disabled": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "button",
        "disabled": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
